lib/linter/apply-disable-directives.js

```diff
@@ -60,34 +60,23 @@ function groupByParentDirective(directives) {
 /**
  * Creates removal details for a set of directives within the same comment.
  * @param {Directive[]} directives Unused directives to be removed.
- * @param {Token} node The backing Comment token.
+ * @param {{node: Token, value: string}} parentDirective Data about the backing directive.
```

Comment:
[Suggestion] Now that parent directives are used more between functions, it might make sense to write up a dedicated `ParentDirective` type for them. They come from `createDisableDirectives > { node, value, ruleIds }` so this inline object type is missing a bit of info.